### PR TITLE
Include webhook data in error notifications

### PR DESCRIPTION
**Pull request by lizconlan**
Connects to mysociety/alaveteli-professional#425

### CODE REVIEW

File: app/controllers/alaveteli_pro/stripe_webhooks_controller.rb [hunk 1]
```diff
@@ -55,15 +55,22 @@ def check_for_event_type
...
   end
 
-  def notify_exception(error)
+  def notify_exception(error, data = nil)
     if send_exception_notifications?
-      ExceptionNotifier.notify_exception(error, :env => request.env)
+      if data
+        ExceptionNotifier.notify_exception(
+          error,
+          data: { webhook_data:  data }
```
Comment (garethrees) [inline app/controllers/alaveteli_pro/stripe_webhooks_controller.rb]: Quick Q: What's the reason for not including `request.env` in this case?

Comment (lizconlan) [reply app/controllers/alaveteli_pro/stripe_webhooks_controller.rb]: oh, er - oversight I think

although I'm not convinced there'll be anything useful in `request.env` in this case (but as the whole point is to have as much information as possible about what might have gone wrong...)

Comment (garethrees) [reply app/controllers/alaveteli_pro/stripe_webhooks_controller.rb]: Okay – I'll give this an initial review but I don't see the point of adding extra conditionals here when we want the `request.env` anyway.

Comment (lizconlan) [reply app/controllers/alaveteli_pro/stripe_webhooks_controller.rb]: agreed, will fix (simplify) post review - thanks!

rough sketch:

```diff
- def notify_exception(error, data = nil)
+ def notify_exception(error, data = {})
...
# remove conditional, always send `data` and `request.env` to notifier
```

**Comment by lizconlan** [comment]
Withdrawing as now that the initial emails are coming through it's obvious that the data _is_ being included

I found this hard to test conclusively without seeing what an actual webhook event looked like and thought it was safer to be sure it was there rather than spending the same amount of time trying to replicate the live environment. In hindsight, taking the time to add an integration test and passing in a mock  event from StripeMock would have been the best approach here.

**Pull request closed without merging**